src/widgets/table/lv_table.c

```diff
@@ -178,16 +178,16 @@ void lv_table_set_cell_value_fmt(lv_obj_t * obj, uint32_t row, uint32_t col, con
     lv_free(raw_txt);
 #else
     table->cell_data[cell] = lv_realloc(table->cell_data[cell],
-                                        sizeof(lv_table_cell_t) + len + 1); /*+1: trailing '\0; */
+                                        sizeof(lv_table_cell_t) + len + 2); /*+1: trailing '\0; */
     LV_ASSERT_MALLOC(table->cell_data[cell]);
     if(table->cell_data[cell] == NULL) {
         va_end(ap2);
         return;
     }
 
-    table->cell_data[cell]->txt[len] = 0; /*Ensure NULL termination*/
+    table->cell_data[cell]->txt[len + 1] = 0; /*Ensure NULL termination*/
 
-    lv_vsnprintf(table->cell_data[cell]->txt, len, fmt, ap2);
+    lv_vsnprintf(table->cell_data[cell]->txt, len + 1, fmt, ap2);
```

Comment:
I've tried out the test code and it seems to me we need to modify only this, to tell that we have space for the `\0` as well.

Reply:
Oh yes,I'll make some modifications